File: articles/decision-under-uncertainty-and-computational-risk/java/DecisionRiskExample.java
public class DecisionRiskExample {
    static double expectedNetValue(double p, double benefit, double loss, double cost) {
        double bounded = Math.max(0.0, Math.min(1.0, p));
        return bounded * benefit - bounded * loss - cost;
    }
    public static void main(String[] args) {
        System.out.printf("%.6f%n", expectedNetValue(0.42, 150.0, 80.0, 25.0));
    }
}

File: articles/decision-under-uncertainty-and-computational-risk/typescript/decision_risk_example.ts
function expectedNetValue(p: number, benefit: number, loss: number, cost: number): number {
  const bounded = Math.max(0, Math.min(1, p));
  return bounded * benefit - bounded * loss - cost;
}

console.log(expectedNetValue(0.42, 150, 80, 25).toFixed(6));
